cmd/cli/plugin/package/test/package_plugin_integration_test.go

```diff
@@ -538,6 +540,17 @@ func pauseKappControllerPackage(clusterName, clusterNamespace, mgmtClusterKubeco
 	Expect(err).ToNot(HaveOccurred())
 }
 
+func disableKctrlCommandTree() {
+	By("disable kctrl command tree")
+	command := exec.NewCommand(
+		exec.WithCommand("tanzu"),
+		exec.WithArgs("config", "set", "features.package.kctrl-package-command-tree", "false"),
```

Comment:
I think modifying [existing test libs](https://github.com/vmware-tanzu/tanzu-framework/blob/main/cmd/cli/plugin/package/test/lib/package_plugin.go#L16) which [package_plugin_integration_test](https://github.com/vmware-tanzu/tanzu-framework/blob/main/cmd/cli/plugin/package/test/package_plugin_integration_test.go) uses would not work, as will break earlier versions tests.

You might want to add your own set of tests for ```kctrl``` feature being enabled.

Reply:
> Until when this flag going to stay false on test?

I am guessing until the old behaviour is deprecated/removed and we no longer need these tests.

> Is it part of the plan to make downstream integration tests eventually pass with kctrl feature enabled as well or if there is another way to for e2e testing kctrl-enabled package

kctrl has it's own set of e2e tests upstream and there are [some tests](https://github.com/vmware-tanzu/tanzu-framework/blob/main/cmd/cli/plugin/package/kctrl/ui_test.go) that make sure that the differences in ux are taken care of. I felt that we would just be duplicating those upstream tests here and hence didn't add any new tests, but please let me know what y'all think.

Reply:
Thanks for the clarification @praveenrewar. As far as I know tests would be needed to get a feature enabled & merged in Tanzu Framework. I let @vijaykatam and @shyaamsn decide on this.

Reply:
@praveenrewar As @maralavi mentioned we cant enable kctrl in tanzu package plugin without tests even though they are tested upstream. We need to test tanzu package plugin in tanzu-framework and downstream.

There are two options:
1. Fix these tests in tanzu-framework so that they can run.
2. Make kctrl tests work in tanzu-framework and downstream

Reply:
Thanks @maralavi and @shyaamsn for sharing your thoughts.
> Fix these tests in tanzu-framework so that they can run.

Sure, I think I can add more tests over here which will work cover the kctrl command tree enabled scenario (the only concern is that I have already mentioned, but I don't mind it)

> Also test-infra e2e tests need to get modified to point to the e2e tests corresponding to kctrl feature after 1.7 release

Which tests are we talking about here, and do we need to make this change as part of the same PR? (Still trying to make my way through the codebase 😅 )

> Make kctrl tests work in tanzu-framework and downstream

Are you referring to the upstream kctrl tests?

Reply:
> Which tests are we talking about here

Hi @praveenrewar , the tests are those in this [file](https://github.com/vmware-tanzu/tanzu-framework/pull/3207/files#diff-fd0938c9729bd8a626a946cab6077d55250011ef9bf024d2abfce781f10e55acR544) which tests all commands end to end, that you are just disabling with your PR.

> do we need to make this change as part of the same PR? (Still trying to make my way through the codebase 😅 )

Yes unfortunately Tanzu Framework code base policy is that everything should be tested properly as part of the same  PR.

Reply:
Ah, okay. That makes sense.